src/praisonai-agents/praisonaiagents/agents/agents.py

```
@@ -2736,6 +2747,199 @@ def close(self) -> None:
+            import threading
+            def _execute_sub_agent():
+                try:
+                    # Execute the task with the sub-agent
+                    if hasattr(task, 'description'):
+                        result = agent.start(task.description)
+                    else:
+                        result = agent.start(str(task))
+                    
+                    # Announce successful completion
+                    self.announce_completion(agent_id, task_id, result, success=True)
+                except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)